### PR TITLE
[Win] Property button

### PR DESCRIPTION
**Pull request by ermau**

### CODE REVIEW

File: Xamarin.PropertyEditing.Windows/PropertyButton.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	internal class PropertyButton
+		: Control
+	{
+		static PropertyButton ()
+		{
+			FocusableProperty.OverrideMetadata (typeof (PropertyButton), new FrameworkPropertyMetadata (false));
+			DefaultStyleKeyProperty.OverrideMetadata (typeof (PropertyButton), new FrameworkPropertyMetadata (typeof (PropertyButton)));
+		}
+
+		public static readonly DependencyProperty CanSetCustomExpressionProperty = DependencyProperty.Register (
```
Comment (bleroy) [inline Xamarin.PropertyEditing.Windows/PropertyButton.cs]: Why are those three properties using `typeof (PropertyEditorControl)` and not `typeof(PropertyButton)`?

File: Xamarin.PropertyEditing.Windows/PropertyButton.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		}
+
+		public override void OnApplyTemplate ()
+		{
+			base.OnApplyTemplate ();
+
+			this.border = (Border)GetTemplateChild ("Border");
+			this.indicator = (Rectangle)GetTemplateChild ("Indicator");
+
+			this.border.MouseDown += OnBorderMouseDown;
```
Comment (bleroy) [inline Xamarin.PropertyEditing.Windows/PropertyButton.cs]: Check for `null`

File: Xamarin.PropertyEditing.Windows/PropertyButton.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			OnValueSourceChanged (ValueSource);
+		}
+
+		private Border border;
+		private Rectangle indicator;
+		private ContextMenu menu;
+
+		private MenuItem systemResources;
+
+		private const double MenuOffset = 3;
```
Comment (bleroy) [inline Xamarin.PropertyEditing.Windows/PropertyButton.cs]: Not used as far as I can tell, should be removed.

File: Xamarin.PropertyEditing.Windows/Themes/Resources.xaml [hunk 4]
```diff
@@ -296,6 +297,57 @@
...
+		<Setter Property="Background" Value="Transparent" />
+		<Setter Property="BorderThickness" Value="0" />
+		<Setter Property="Height" Value="17" />
+		<Setter Property="Width" Value="11" />
+		<Setter Property="SnapsToDevicePixels" Value="True" />
+		<Setter Property="MenuTemplate">
+			<Setter.Value>
+				<DataTemplate>
+					<local:HeaderedContextMenu Header="{Binding Property.Name,Mode=OneTime}" StaysOpen="True" Placement="Left" HorizontalOffset="{StaticResource MenuOffset}" VerticalOffset="{StaticResource MenuOffset}">
+						<!--<MenuItem x:Name="CustomExpressionItem" Header="{x:Static prop:Resources.CustomExpressionEllipsis}" Icon="{StaticResource LiteralMarker}" />
```
Comment (bleroy) [inline Xamarin.PropertyEditing.Windows/Themes/Resources.xaml]: Any particular reason to keep commented out code around?

File: Xamarin.PropertyEditing.Windows/Themes/VS.Light.xaml [hunk 5]
```diff
@@ -42,6 +42,8 @@
 	<SolidColorBrush x:Key="ListItemDisabledForegroundBrush">#FFA2A4A5</SolidColorBrush>
 	<SolidColorBrush x:Key="MenuSeparatorBrush">#E0E3E6</SolidColorBrush>
 
+	<SolidColorBrush x:Key="LiteralMarkerBrush">#55000000</SolidColorBrush>
```
Comment (bleroy) [inline Xamarin.PropertyEditing.Windows/Themes/VS.Light.xaml]: You don't seem to have done the dark version of that.

File: Xamarin.PropertyEditing.Windows/Themes/Resources.xaml (hunk 4 above)
Comment (ermau) [reply Xamarin.PropertyEditing.Windows/Themes/Resources.xaml]: We'll be adding them back as we have support for them.

**Pull request merged**